resilience4j-spring/src/main/java/io/github/resilience4j/bulkhead/configure/PlainObjectBulkheadAspectExt.java

```diff
@@ -0,0 +1,139 @@
+
+    public PlainObjectBulkheadAspectExt(TimeLimiterRegistry timeLimiterRegistry) {
+        this.timeLimiterRegistry = timeLimiterRegistry;
+        this.executorService = Executors.newScheduledThreadPool(Runtime.getRuntime().availableProcessors());
+    }
+
+    /**
+     * Determines if the aspect can handle the specified return type of the method.
+     *
+     * @param returnType the AOP method return type class
+     * @return {@code true} if the method has a plain object return type; {@code false} otherwise.
+     */
+    @Override
+    @SuppressWarnings("rawtypes")
+    public boolean canHandleReturnType(Class returnType) {
+        return !returnType.equals(CompletableFuture.class)
+                && !returnType.equals(Mono.class)
+                && !returnType.equals(Flux.class);
+    }
+
+    /**
+     * Handle the execution of a method wrapped in a bulkhead context.
+     *
+     * @param proceedingJoinPoint Spring AOP proceedingJoinPoint
+     * @param bulkhead the configured bulkhead
+     * @param methodName the method name
+     * @return the result object from the method execution
+     * @throws Throwable if an exception occurs during the method execution
+     */
+    @Override
+    public Object handle(ProceedingJoinPoint proceedingJoinPoint, Bulkhead bulkhead, String methodName) throws Throwable {
+        if (bulkhead instanceof ThreadPoolBulkheadAdapter) {
+            ThreadPoolBulkhead threadPoolBulkhead = ((ThreadPoolBulkheadAdapter) bulkhead).threadPoolBulkhead();
+            return handleThreadPoolBulkhead(proceedingJoinPoint, threadPoolBulkhead, methodName);
+        }
+
+        return handleSemaphoreBulkhead(proceedingJoinPoint, bulkhead, methodName);
+    }
+
+    private Object handleSemaphoreBulkhead(ProceedingJoinPoint proceedingJoinPoint, Bulkhead bulkhead, String methodName) {
+        TimeLimiter timeLimiter = timeLimiterRegistry.timeLimiter(bulkhead.getName());
+        try {
+            Supplier<CompletableFuture<Object>> futureSupplier = createBulkheadFuture(proceedingJoinPoint, bulkhead);
+            return timeLimiter.executeCompletionStage(executorService, futureSupplier)
+                    .toCompletableFuture()
+                    .join();
+        } catch (BulkheadFullException ex) {
+            logBulkheadFullException(methodName, ex);
+            throw ex;
+        } catch (CompletionException ex) {
+            logCompletionException(methodName, ex);
+            throw ex;
+        } catch (Throwable ex) {
+            logGenericException(methodName, ex);
+            throw ex;
+        }
+    }
+
+    private Object handleThreadPoolBulkhead(ProceedingJoinPoint proceedingJoinPoint, ThreadPoolBulkhead threadPoolBulkhead, String methodName) {
+        TimeLimiter timeLimiter = timeLimiterRegistry.timeLimiter(threadPoolBulkhead.getName());
+        try {
+            CompletableFuture<Object> completableFuture = threadPoolBulkhead.executeCallable(() -> {
```

Comment:
This is wrong.
It already executes the call before you have wrapped it with a TimeLimiter.
Use `decorateCallable` instead